lms/djangoapps/onboarding/helpers.py

```diff
@@ -8010,3 +8011,24 @@ def serialize_partner_networks():
         ))
 
     return data
+
+
+def get_user_on_demand_courses(user):
+    all_user_courses = CourseEnrollment.objects.filter(user=user, is_active=True)
+    courses = []
+    for user_course in all_user_courses:
+        try:
+            courses.append(CourseOverview.objects.get(id=user_course.course_id, self_paced=True))
+        except CourseOverview.DoesNotExist:
+            continue
+    return courses
+
+
+def get_email_pref_first_on_demand_courses(user, on_demand_courses=None):
```

Comment:
Let's please make it generic, this will take a single `course` and a `user` and return email_preference of the user for that course.
Also function documentation please.